packages/protocol/test/governance/network/blockchainparams.ts

```diff
@@ -114,30 +125,144 @@ contract('BlockchainParameters', (accounts: string[]) => {
     })
   })
 
+  describe('#getUptimeLookbackWindow()', () => {
+    it('should return 12 if not initalized', async () => {
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 12)
+    })
+
+    it('should return oldValue once it has been initialized', async () => {
+      await blockchainParameters.setUptimeLookbackWindow(20)
+      await mineBlocks(EPOCH, web3)
+      await blockchainParameters.setUptimeLookbackWindow(50)
+
+      // returns old value
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 20)
+    })
+  })
+
+  describe('#setUptimeLookbackWindow()', () => {
+    const lookbackWindow = 20
+
+    it('if set should set the variable for the next epoch', async () => {
+      // enter steady state
+      await blockchainParameters.setUptimeLookbackWindow(24)
+      await mineBlocks(EPOCH, web3)
+
+      await blockchainParameters.setUptimeLookbackWindow(40)
+
+      // still old value
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+
+      // wait an epoch to find new value
+      await mineBlocks(EPOCH, web3)
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 40)
+    })
+
+    it('if unset should set the variable for the current epoch', async () => {
```

Comment:
I think this would benefit in readability if these pre-conditions are set with sub-describe's and beforeEach's